src/common/dpar_mpi.c

```diff
@@ -0,0 +1,210 @@
+#include <stdio.h>
+#include <mpi.h>
+#include <daos/dpar.h>
+
+uint32_t
+par_getversion(void)
+{
+	return DPAR_VERSION;
+}
+
+int
+par_init(int *argc, char ***argv)
+{
+	int rc = MPI_Init(argc, argv);
+
+	if (rc == MPI_SUCCESS)
+		return 0;
+
+	fprintf(stderr, "MPI_Init failed with %d\n", rc);
+	return -1;
+}
+
+int
+par_fini(void)
+{
+	int rc = MPI_Finalize();
+
+	if (rc == MPI_SUCCESS)
+		return 0;
+
+	fprintf(stderr, "MPI_Finalize failed with %d\n", rc);
+	return -1;
+}
+
+int
+par_barrier(void)
+{
+	int rc = MPI_Barrier(MPI_COMM_WORLD);
+
+	if (rc == MPI_SUCCESS)
+		return 0;
+
+	fprintf(stderr, "MPI_Barrier failed with %d\n", rc);
+	return -1;
+}
+
+int
+par_rank(int *rank)
```

Comment:
Since we didn't have any usage where we do anything but MPI_COMM_WORLD, I did it this way.  I can revisit.  Maybe in a subsequent patch?  This one has been around for ages waiting for 2.2

Reply:
Or perhaps a separate API.  It's a tricky one because then we would have to wrap a bunch of APIs that create them.  And MPI_Comm isn't the same size across MPI versions (32-bit for mpich and 64-bit for openmpi).  Anyway, I can figure out a way but since it's not needed for the immediate issue, it seems reasonable to defer to later.

Reply:
Nevermind, I went ahead and added it.